### PR TITLE
[For Discussion] Endpoint Example

### PR DESCRIPTION
**Pull request by TammoR**
Here is an example of how the client functions would look like.

**Thought**
I think we should still have a separate set of endpoints (just thin wrappers). - Because we may want to expose a different set of arguments in the client than for our internal endpoints. E.g. we may want to access projects or test cases by a slug, which is not currently part of our endpoint.

See inline comments for an example.

**Alternatives**
(1) Add any required parameters to our existing internal endpoints. We can always only offer a subset of parameters in the client. (E.g. don't let users provide an "id" which is never exposed to them).
(2) Just keep going with what we have, built out the Fini use-case MVP, and discuss then.

### CODE REVIEW

File: tests/test_dataset.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import os
+
+# Assuming `SampleClient` has a method `create_test_case_dataset`
+# and it is defined in a module called `client_module`
+from ivycheck.client import IvyClient
+
+# Set up your API key and base URL
+ivy = IvyClient(api_key=os.environ["IVYCHECK_API_KEY"])
+
+# Call the `create_test_case_dataset` method with dummy arguments
```
Comment (TammoR) [inline tests/test_dataset.py:10]: So below you see how one would call the endpoint from the client. We can simply use all available parameters as optional kwargs for greatest convenience. Problems:
- `prompt_id` should actually be `project_id` but we should probably rename this everywhere
- More interestingly: We probably don't want to use the id here. Users won't know the prompt id. They might know the slug. (Associating test datasets with prompts is not a main use case anyways). A more relevant occurrence of the same issue: We should give slugs to test cases and test case datasets. This is something that users will refer to often.

Comment (lange-labs) [reply tests/test_dataset.py:10]: - Re prompt_id: nice thing about the client is that we can just rename it there right now and do the transformation to whatever it is in the backend separately later.
- Re id in general: good point re using slug for project, but maybe let's check for a sec what people usually do in other apps. There are some pros and cons to using slugs: easier to read, but you need to come up with unique names on the fly. Also, are they required vs. can you also use the id instead.

File: ivycheck/client.py [hunk 2]
```diff
@@ -59,3 +68,40 @@ def check_endpoint_health(self):
...
+            response.raise_for_status()
+
+    def delete_test_dataset():
+        pass
+
+    def update_test_dataset():
+        pass
+
+    def read_test_dataset():
+        pass
```
Comment (lange-labs) [inline ivycheck/client.py:107]: 1. For the next functions to implement, shall we do the minimum required for the Fini-like MVP?
2. Packages: should the functions all be in a single main package or should there be multiple sub packages like test_dataset etc.? We could also look at some "famous" APIs like stripe.

Comment (lange-labs) [reply ivycheck/client.py:107]: Looks good! 

Just looked at stripe at it looks similar, but using camel case:

Screenshot 2023-11-21 at 08 56 04

**Comment by TammoR** [comment]
Closing in favor of new PR

**Pull request closed without merging**